test/parallel/test-process-env.js

```diff
@@ -59,3 +59,7 @@ if (process.argv[2] == 'you-are-the-child') {
     }
   });
 }
+
+// delete should return true except for non-configurable properties
+// https://github.com/nodejs/node/issues/7960
+assert(delete process.env.NON_EXISTING_VARIABLE);
```

Comment:
Also, I'd do an initial `delete process.env.NON_EXISTING_VARIABLE` just in case someone has NON_EXISTING_VARIABLE set in their environment.